### PR TITLE
feat(xcode): only install xcode if it doesn't already exist

### PR DESCRIPTION
**Pull request by rimarsh**
I don't install Xcode from the App Store, so this is letting it skip if Xcode already exists. What do you think?

### CODE REVIEW

File: plugins/xcode/xcode [hunk 1]
```diff
@@ -3,6 +3,18 @@
...
+  else
+    _install_xcode_from_mas
+  fi
+
+  debug 'Making sure we have accepted the xcode license'
+  sudo xcodebuild -license accept
+}
+
+_install_xcode_from_mas() {
+  mv XcodeBrewfile Brewfile
```
Comment (rimarsh) [inline plugins/xcode/xcode:17]: This needs to be updated to not be relative

Comment (ianwremmel) [reply plugins/xcode/xcode:17]: But what's it trying to do?

Comment (ianwremmel) [reply plugins/xcode/xcode:17]: oh, i see. I completely missed the file rename above.

**Review by ianwremmel — APPROVED** [review]
I think this makes sense, though can you add some comments on what it's doing? Also, what happens if someone wants to switch back to the app store version?

File: plugins/xcode/xcode (hunk 1 above)
Comment (rimarsh) [reply plugins/xcode/xcode:17]: Yeah, I'm open to other ides for how to make it skip this Brewfile if you prefer.

File: plugins/xcode/xcode [hunk 2]
```diff
@@ -3,6 +3,27 @@
 export DOTFILES_XCODE_DEPS=('homebrew_core')
 
 dotfiles_xcode_apply () {
+  # Sometimes it's preferrable to have Xcode installed from
```
Comment (ianwremmel) [inline plugins/xcode/xcode:6]: why are you wrapping your comment blocks at 65 characters? :)

File: plugins/xcode/xcode [hunk 3]
```diff
@@ -3,6 +3,27 @@
...
+    _install_xcode_from_mas
+  fi
+
+  debug 'Make sure we have accepted the xcode license'
+  sudo xcodebuild -license accept
+}
+
+_install_xcode_from_mas() {
+  local xcode_plugin_path
+  xcode_plugin_path=$(plugin_get_path xcode)
```
Comment (ianwremmel) [inline plugins/xcode/xcode:22]: neat. totally forgot that function existed.

File: plugins/xcode/xcode [hunk 4]
```diff
@@ -3,6 +3,27 @@
...
+  
+  # Copy Brewfile for Xcode into expected place
+  cp $xcode_plugin_path/XcodeBrewfile $xcode_plugin_path/Brewfile
+
```
Comment (ianwremmel) [inline plugins/xcode/xcode:26]: please run shellcheck, I think you're missing a bunch of quotes.

Comment (ianwremmel) [reply plugins/xcode/xcode:26]: there's a vscode plugin that should Just Work (TM)

File: plugins/xcode/xcode (hunk 1 above)
Comment (ianwremmel) [reply plugins/xcode/xcode]: nah, this makes sense, I just missed the rename during the code review, so I was very confused.

**Review by ianwremmel — APPROVED** [review]

**Pull request merged**